dashboard/legacy/middleware/helpers/projects.rb

```diff
@@ -13,12 +13,23 @@ class NotFound < Sinatra::NotFound
   class ValidationError < StandardError
   end
 
+  class PublishError < StandardError
+  end
+
   def initialize(storage_id)
     @storage_id = storage_id
 
     @table = Projects.table
   end
 
+  #### NOTE: This references the Rails model (Project, singular)
+  #### rather than this middleware class (Projects, plural)
+  #### such that we can make use of model associations managed by Rails.
+  def get_rails_project(project_id)
```

Comment:
Especially curious for thoughts on this -- I think now that the `projects` table is in the dashboard database, we can start moving over project logic (and routing) into more standard Rails approaches. I'm hoping that this is a good starting point for that.